src/cryptonote_basic/cryptonote_basic_impl.cpp

```diff
@@ -310,6 +310,42 @@ namespace cryptonote {
   bool operator ==(const cryptonote::block& a, const cryptonote::block& b) {
     return cryptonote::get_block_hash(a) == cryptonote::get_block_hash(b);
   }
+  //--------------------------------------------------------------------------------
+  int compare_hash32_reversed_nbits(const crypto::hash& ha, const crypto::hash& hb, unsigned int nbits)
+  {
+    const uint64_t* const va = reinterpret_cast<const uint64_t*>(ha.data);
+    const uint64_t* const vb = reinterpret_cast<const uint64_t*>(hb.data);
+
+    for (int n = 3; n >= 0 && nbits; --n)
```

Comment:
`static_assert` that `sizeof(uint64_t) * 4 == sizeof(crypto::hash)`. I doubt we'll ever change this, but its a nice catch regardless.